### PR TITLE
[Compute] sig image-version create: Support creating from a VHD

### PR DESCRIPTION
**Pull request by qwordy**
**Description**

Resolve https://github.com/Azure/azure-cli/issues/15868

[Compute] Upgrade gallery_image_versions to 2020-09-30
[Compute] sig image-version create: Support creating from a VHD

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -3348,6 +3351,17 @@ def create_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
             for i, s in enumerate(data_snapshots):
                 data_disk_images.append(GalleryDataDiskImage(source=GalleryArtifactVersionSource(id=s),
                                                              lun=data_snapshot_luns[i]))
+        # from vhd, only support os image now
+        if cmd.supported_api_version(min_api='2020-09-30', operation_group='gallery_image_versions'):
+            if vhd and vhd_storage_account is None or vhd is None and vhd_storage_account:
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:3356]:
```suggestion
            if not all([vhd, vhd_storage_account]):
```

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -897,6 +897,8 @@ def load_arguments(self, _):
                    arg_type=get_enum_type(["Standard_LRS", "Standard_ZRS", "Premium_LRS"]), min_api='2019-03-01')
         c.argument('target_region_encryption', nargs='+',
                    help='Space-separated list of customer managed keys for encrypting the OS and data disks in the gallery artifact for each region. Format for each region: `<os_des>,<lun1>,<lun1_des>,<lun2>,<lun2_des>`. Use "null" as a placeholder.')
+        c.argument('vhd', help='Source VHD URI of OS disk', min_api='2020-09-30')
+        c.argument('vhd_storage_account', help='Name or ID of storage account of source VHD URI of OS disk', min_api='2020-09-30')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:901]: could vhd resource in another tenant?

**Review by jiasli — APPROVED** [review]

**Review by fengzhou-msft — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 2 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:901]: Good question. If true, I need to add cross-tenant authentication.

**Comment by olayemio** [comment]
I have a few questions looking at the api support. So a customer using VHD will supply vhd and vhd_storage_account flag, but not the os_snapshot flag?
1. Would it be possible for customers to use the os_snapshot field for the ID of the storage account, and then a separate argument for the URI of the VHD for the OS disk?
2. Is there support for data disks to use VHD as well? data_snapshots could be used to specify the ID of the storage accounts and then a separate argument for the URI of all the VHDs

**Comment by yungezz** [comment]
hi @qwordy, what's status of this PR? is it ok to merge?

**Comment by qwordy** [comment]
1. Agree. The arguments can be better.
2. We should also support data disks.
I plan to merge OS disk part, than put data disks in another PR.

**Comment by olayemio** [comment]
@qwordy 
If you want some suggestions on the arguments, here's something I'm thinking
- for OS disk, can use `--os-snapshot-uri` for the VHD uri
- for data disks, since we need the storage account id and VHD uri, `--data-snapshots` can be used for the storage account, and the VHD can be specified using `--data-vhd` and `--data-vhd-luns` to indicate which uri match up with which storage account

**Comment by qwordy** [comment]
@olayemio How about `--os-vhd`, `--os-vhd-storage-account`, `--data-vhd`, `--data-vhd-luns`, `--data-vhd-storage-accounts`? Can we omit storage account? A VHD URI is unique.

**Review by houk-ms — APPROVED** [review]

**Comment by qwordy** [comment]
The OS part is done.

**Comment by olayemio** [comment]
@qwordy I propose the following `--os-vhd-uri`, `--os-vhd-storage-account`, `--data-vhds-uri`, `--data-vhds-luns`, `--data-vhds-storage-accounts`? 
The storage account is required to ensure that there is a linked access check for the VHD

**Comment by qwordy** [comment]
Merge OS part first. Data disks will be in another PR.

**Comment by qwordy** [comment]
https://github.com/Azure/azure-cli/pull/17706

**Pull request merged**